core/src/avm2/globals/date.rs

```diff
@@ -3,15 +3,61 @@
 use crate::avm2::activation::Activation;
 use crate::avm2::class::Class;
 use crate::avm2::method::{Method, NativeMethodImpl};
-use crate::avm2::names::{Namespace, QName};
-use crate::avm2::object::{date_allocator, DateObject, Object, TObject};
+use crate::avm2::names::{Multiname, Namespace, QName};
+use crate::avm2::object::{date_allocator, DateObject, FunctionObject, Object, TObject};
 use crate::avm2::value::Value;
 use crate::avm2::Error;
 use crate::string::{utils as string_utils, AvmString, WStr};
 use chrono::{DateTime, Datelike, Duration, FixedOffset, LocalResult, TimeZone, Timelike, Utc};
 use gc_arena::{GcCell, MutationContext};
 use num_traits::ToPrimitive;
 
+// All of these methods will be defined as both
+// AS3 instance methods and methods on the `Date` class prototype.
+const PUBLIC_INSTANCE_AND_PROTO_METHODS: &[(&str, NativeMethodImpl)] = &[
+    ("getTime", time),
+    ("setTime", set_time),
+    ("getMilliseconds", milliseconds),
+    ("setMilliseconds", set_milliseconds),
+    ("getSeconds", seconds),
+    ("setSeconds", set_seconds),
+    ("getMinutes", minutes),
+    ("setMinutes", set_minutes),
+    ("getHours", hours),
+    ("setHours", set_hours),
+    ("getDate", date),
+    ("setDate", set_date),
+    ("getMonth", month),
+    ("setMonth", set_month),
+    ("getFullYear", full_year),
+    ("setFullYear", set_full_year),
+    ("getDay", day),
+    ("getUTCMilliseconds", milliseconds_utc),
+    ("setUTCMilliseconds", set_milliseconds_utc),
+    ("getUTCSeconds", seconds_utc),
+    ("setUTCSeconds", set_seconds_utc),
+    ("getUTCMinutes", minutes_utc),
+    ("setUTCMinutes", set_minutes_utc),
+    ("getUTCHours", hours_utc),
+    ("setUTCHours", set_hours_utc),
+    ("getUTCDate", date_utc),
+    ("setUTCDate", set_date_utc),
+    ("getUTCMonth", month_utc),
+    ("setUTCMonth", set_month_utc),
+    ("getUTCFullYear", full_year_utc),
+    ("setUTCFullYear", set_full_year_utc),
+    ("getUTCDay", day_utc),
+    ("getTimezoneOffset", timezone_offset),
+    ("valueOf", time),
+    ("toString", to_string),
+    ("toUTCString", to_utc_string),
+    ("toLocaleString", to_locale_string),
+    ("toTimeString", to_time_string),
+    ("toLocaleTimeString", to_locale_time_string),
+    ("toDateString", to_date_string),
+    ("toLocaleDateString", to_date_string),
+];
```

Comment:
to be sure: no `toJSON`? It's okay if it's not in this PR, just making sure.

Reply:
It looks like we don't currently define `toJSON` at all, so I think it would be better to leave it for a follow-up PR.